electron/SafeLogger.ts

```
@@ -0,0 +1,111 @@
+import fs from 'fs'
+import path from 'path'
+import { app } from 'electron'
+
+class SafeLogger {
+  private logDir: string
+  private mainLogPath: string
+  private shortcutLogPath: string
+
+  constructor() {
+    // Initialize log directory
+    this.logDir = path.join(app.getPath('userData'), 'logs')
+    this.mainLogPath = path.join(this.logDir, 'main-events.log')
+    this.shortcutLogPath = path.join(this.logDir, 'shortcut-events.log')
+
+    // Create logs directory if it doesn't exist
+    this.ensureLogDirectory()
+  }
+
+  private ensureLogDirectory(): void {
+    try {
+      if (!fs.existsSync(this.logDir)) {
+        fs.mkdirSync(this.logDir, { recursive: true })
+      }
+    } catch (error) {
+      // Silently fail - we can't log the error since we're creating the logger
```

I don't think we should silently fail this - if this fails there is no way of figuring out what went wrong before or after